clang/test/SemaSYCL/spec-const-value-dependent-crash.cpp

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -ast-dump -sycl-std=2020 %s | FileCheck %s
+// This test checks that Clang doesn't crash if a specialization constant is
+// value dependent.
+
+#include "sycl.hpp"
+sycl::queue myQueue;
+
+int main() {
+   constexpr int default_val = 20;
+   cl::sycl::ext::oneapi::experimental::spec_constant<int, class MyInt32Const> SC(default_val);
```

Comment:
@srividya-sundaram can you please re-write the test to use SYCL 2020 spec constants, and not outdated oneAPI extension for spec constants?
As an example, sycl/test/basic_tests/SYCL-2020-spec-const-ids-order.cpp will be good to see the basic syntax.

FYI @AlexeySachkov